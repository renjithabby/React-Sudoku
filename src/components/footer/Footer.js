import React from 'react';
import PropTypes from 'prop-types';

import './style.css';

const Footer =  ({validEntries,selectedInput,setSelectedInputValue}) => {
    const onButtonClick = (event)=>{
        setSelectedInputValue(parseInt(event.target.value));
    }
    return (
        <div className = 'footer-container'>
            <span>Please choose an input value first</span>
            <div className = 'footer-buttons'>
                {validEntries.map(value =>{
                    return (<button className={`footer-button ${selectedInput===value?'footer-button--selected':''}`} key={value} onClick={onButtonClick} value={value} > {value}</button>)
                })}
            </div>
        </div>
    );
};

export default Footer;

Footer.propTypes= {
    validEntries : PropTypes.array,
    selectedInput : PropTypes.number,
    setSelectedInputValue : PropTypes.func.isRequired,
}

Footer.defaultProps= {
    validEntries  : [1,2,3,4,5,6,7,8,9]
}