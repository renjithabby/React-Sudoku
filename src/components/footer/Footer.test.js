import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Footer from './Footer';

configure({adapter : new Adapter()});

describe.only('<Footer />', () => {
    
    it('It matches the snapshot', () => {
        const tree = shallow(<Footer setSelectedInputValue={()=>{}}/>);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('It matches the snapshot with props selectedInput value', () => {
        const tree = shallow(<Footer selectedInput={5} setSelectedInputValue={()=>{}}/>);
        expect(toJson(tree)).toMatchSnapshot();
    });

})