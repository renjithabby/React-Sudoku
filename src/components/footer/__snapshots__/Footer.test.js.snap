// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> It matches the snapshot 1`] = `
<div
  className="footer-container"
>
  <span>
    Please choose an input value first
  </span>
  <div
    className="footer-buttons"
  >
    <button
      className="footer-button "
      key="1"
      onClick={[Function]}
      value={1}
    >
       
      1
    </button>
    <button
      className="footer-button "
      key="2"
      onClick={[Function]}
      value={2}
    >
       
      2
    </button>
    <button
      className="footer-button "
      key="3"
      onClick={[Function]}
      value={3}
    >
       
      3
    </button>
    <button
      className="footer-button "
      key="4"
      onClick={[Function]}
      value={4}
    >
       
      4
    </button>
    <button
      className="footer-button "
      key="5"
      onClick={[Function]}
      value={5}
    >
       
      5
    </button>
    <button
      className="footer-button "
      key="6"
      onClick={[Function]}
      value={6}
    >
       
      6
    </button>
    <button
      className="footer-button "
      key="7"
      onClick={[Function]}
      value={7}
    >
       
      7
    </button>
    <button
      className="footer-button "
      key="8"
      onClick={[Function]}
      value={8}
    >
       
      8
    </button>
    <button
      className="footer-button "
      key="9"
      onClick={[Function]}
      value={9}
    >
       
      9
    </button>
  </div>
</div>
`;

exports[`<Footer /> It matches the snapshot with props selectedInput value 1`] = `
<div
  className="footer-container"
>
  <span>
    Please choose an input value first
  </span>
  <div
    className="footer-buttons"
  >
    <button
      className="footer-button "
      key="1"
      onClick={[Function]}
      value={1}
    >
       
      1
    </button>
    <button
      className="footer-button "
      key="2"
      onClick={[Function]}
      value={2}
    >
       
      2
    </button>
    <button
      className="footer-button "
      key="3"
      onClick={[Function]}
      value={3}
    >
       
      3
    </button>
    <button
      className="footer-button "
      key="4"
      onClick={[Function]}
      value={4}
    >
       
      4
    </button>
    <button
      className="footer-button footer-button--selected"
      key="5"
      onClick={[Function]}
      value={5}
    >
       
      5
    </button>
    <button
      className="footer-button "
      key="6"
      onClick={[Function]}
      value={6}
    >
       
      6
    </button>
    <button
      className="footer-button "
      key="7"
      onClick={[Function]}
      value={7}
    >
       
      7
    </button>
    <button
      className="footer-button "
      key="8"
      onClick={[Function]}
      value={8}
    >
       
      8
    </button>
    <button
      className="footer-button "
      key="9"
      onClick={[Function]}
      value={9}
    >
       
      9
    </button>
  </div>
</div>
`;
